viewing_party/main.py

```diff
@@ -0,0 +1,181 @@
+### --------------- WAVE 1 --------------- ###
+
+def create_movie(movie_title, genre, rating):
+    # take in three parameters title, genre, rating
+    new_movie = {
+        "title": movie_title,
+        "genre": genre,
+        "rating":  rating
+    }
+    # if title, genre, OR rating is falsy, return None
+    for values in new_movie.values():
+        if values == None:
+            return None
+    # step 3: if the attributes are truthy, return dictionary
+    else:
+        return new_movie
+
+
+def add_to_watched(user_data, movie):
+    # access the value of "watched" list and assign variable
+    watched_list = user_data["watched"]
+
+    # append movie to "watched" list
+    watched_list.append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    # just repeat what you did in add_to_watched
+    watchlist = user_data["watchlist"]
+    watchlist.append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    # iterate thru the movies in "watchlist" list
+    for movie in user_data['watchlist']:
+        # check if the value of movie["title"] matches the paremeter title 
+        if title == movie['title']:
+            # then add to "watched" list and remove from "watchedlist" list
+            user_data['watched'].append(movie)
+            user_data['watchlist'].remove(movie)
+    return user_data
+
+### --------------- WAVE 2 --------------- ###
+
+def get_watched_avg_rating(user_data):
+    watched_list = user_data["watched"]
+    rating_list = []
+
+    # loop to make list that captures only the ratings
+    for movie in watched_list:
+        rating_list.append(movie["rating"])
+    # handle for zero division error
+    if len(rating_list) == 0:
+        avg_rating = 0.0
+        return avg_rating
+    else:
+        avg_rating = sum(rating_list)/len(rating_list)
+        return avg_rating
```

Comment:
so, this for loop will run whether or not it's needed. let's try checking `len(rating_list)` first:

```suggestion
if len(watched_list) > 0:
    avg_rating = 0.0
    for movie in watched_list:
        avg_rating += watched_list["rating"]
    avg_rating = avg_rating / len(watched_list)
    
return avg_rating
```

now it won't run lines of code it doesn't need! also, the function `sum` runs a loop behind the scenes in order to work, so that means you are using two for-loops. we can have the first one do the work of both!